numpy/core/numeric.py

```diff
@@ -401,11 +402,17 @@ def asarray(a, dtype=None, order=None):
         Input data, in any form that can be converted to an array.  This
         includes lists, lists of tuples, tuples, tuples of tuples, tuples
         of lists and ndarrays.
-    dtype : data-type, optional
+    dtype : {None, data-type}, optional
         By default, the data-type is inferred from the input data.
-    order : {'C', 'F'}, optional
-        Whether to use row-major ('C') or column-major ('F' for FORTRAN)
-        memory representation.  Defaults to 'C'.
+    order : {'C', 'F', 'A'}, optional
+        Whether to use row-major ('C'), column-major ('F'), or any ('A')
+        memory representation.  Defaults to 'A'.
+    precision: {None, dtype, ndarray, scalar}, optional
```

Comment:
I agree, what the scalar does is difficult to understand, but the `inexact=True` logic seems actually the most common one? And it is not quite the same as *any* dtype. I once thought about a prepare_arrays (something like asarray but which takes `*arrays` and would do things like this, plus give the common result type/cast them all) and I did not even think about min_dtype, but only about `inexact`.
Was thinking to do that in C, but could sketch up a python implementation of something like this to give an idea if that seems like it might be an option.

You do not need to mention scalars ;), since they are actually cast to 0-d arrays in `result_type`, but that does *not* fix the issue that the usage, which to me currently seems the most useful, is completely hidden away in `scalar` which makes no sense without more explanation. (though it always needs explanation). I don't have a solution, but hopefully these are not just unhelpful random thoughts :).